src/backend/utils.ts

```diff
@@ -915,19 +936,79 @@ async function ContinueWithFoundWine(
   return { response }
 }
 
+export async function isRosettaAvailable() {
+  if (!isMac) {
+    return
+  }
+
+  // check if on arm64 macOS
+  const { stdout: archCheck } = await execAsync('arch')
+  const isArm64 = archCheck.trim() === 'arm64'
+
+  if (!isArm64) {
+    return
+  }
+
+  const { stdout: rosettaCheck } = await execAsync(
+    'arch -x86_64 /usr/sbin/sysctl sysctl.proc_translated'
+  )
+
+  const result = rosettaCheck.split(':')[1].trim() === '1'
+
+  logInfo(
+    `Rosetta is ${result ? 'available' : 'not available'} on this system.`,
+    LogPrefix.Backend
+  )
+
+  if (!result) {
+    // show a dialog saying that hyperplay wont run without rosetta and add information on how to install it
+    await dialog.showMessageBox({
```

Comment:
This looks odd, we're showing a warning dialog in a function that checks if rosseta is available.
Can we move it outside?

Reply:
Yeah, true. I will think on how to do that, maybe just rename the method.
Because actually, we are not using the return here for anything, its just to check and show the warning.

Reply:
done